samples/csharp/getting-started/Ranking_PersonalizedSort/README.md

```diff
@@ -0,0 +1,215 @@
+5. The final step is to **consume** the model to perform ranking predictions for new incoming hotel searches.
+
+### 1. Label Data
+To label the data with ideal ranking values, the sample follows [Expedia's evaluation guidance](https://www.kaggle.com/c/expedia-personalized-sort/overview/evaluation):
+
+* 0 - The user neither clicked on this hotel nor purchased\booked a room at this hotel.
+* 1 - The user clicked through to see more information on this hotel.
+* 2 - The user purchased\booked a room at this hotel.
+
+Expedia's dataset includes both **Click_Bool** and **Booking_Bool** columns that indicate whether the user has clicked or purchased/booked a hotel.  Applying the above guidelines to these columns, we create a new **Label** column that contains values {0, 1, 2} for each hotel search result.
+
+The code for labeling the data is similar to the following:
+
+```CSharp
+// Load dataset using TextLoader by specifying the type name that holds the data's schema to be mapped with datasets.
+IDataView data = mlContext.Data.LoadFromTextFile<HotelData>(originalDatasetPath, separatorChar: ',', hasHeader: true);
+
+// Create an Estimator and use a custom mapper to transform label hotel instances to values 0, 1, or 2.
+IEstimator<ITransformer> dataPipeline = mlContext.Transforms.CustomMapping(Mapper.GetLabelMapper(mlContext, data), null);
+
+// To transform the data, call the Fit() method.
+ITransformer dataTransformer = dataPipeline.Fit(data);
+IDataView labeledData = dataTransformer.Transform(data);
+
+[...]
+
+// Custom mapper used to label a hotel search result with the ideal rank. 
+public static Action<HotelData, HotelRelevance> GetLabelMapper(MLContext mlContext, IDataView data)
+{
+    Action<HotelData, HotelRelevance> mapper = (input, output) =>
+    {
+        if (input.Srch_Result_Booked == 1)
+        {
+            output.Label = 2;
+        }
+        else if (input.Srch_Result_Clicked == 1)
+        {
+            output.Label = 1;
+        }
+        else
+        {
+            output.Label = 0;
+        }
+    };
+
+    return mapper;
+}
+`````
+### 2. Split Data
+ With the data properly labeled, it is ready to be split into the train/test datasets.  When splitting the data, it's important to make sure that the hotel search results for a given query aren't split across the two datasets.  This would cause label leakage where the same query in our training dataset also exists within the testing dataset.
+
+ Refer to the following code which shows how to split the data:
+
+ ```CSharp
+// When splitting the data, 20% is held for the test dataset.
+// To avoid label leakage, the GroupId (e.g. search\query id) is specified as the samplingKeyColumnName.  
```

Comment:
```suggestion
// To avoid data leakage, the GroupId (e.g. search\query id) is specified as the samplingKeyColumnName.  
```

Reply:
Note that the actual documentation provided for this parameter says the following: "This can be used to ensure no label leakage from the train to the test set."

This obviously is a minor issue in the doc, but sounds like the way it's currently written is incorrect?

Reply:
#resolved